File: 서인/CH02/2_보이는학생.js
// 앞 인덱스의 value보다 작거나 같은 index 카운트 안함

function solution(arr) {
    let answer = 0;
    let max = Number.MIN_SAFE_INTEGER;
  
    for (let x of arr) {
      if (x > max) {
        answer++;
        max = x;
      }
    }
  
    return answer;
  }
  
  let arr = [130, 135, 148, 140, 145, 150, 150, 153];
  console.log(solution(arr));
  

File: 종복/Java/Array/2. 보이는 학생.java
import java.util.*;
class Main {	
	public int solution(int n, int[] arr){
		int answer=1;
        int max = arr[0];
        
        for(int i=1; i<arr.length; i++){
             if(max< arr[i]){
               max = arr[i];
               answer += 1;
             }
        }
		
        return answer; 
	}
	public static void main(String[] args){
		Main T = new Main();
		Scanner kb = new Scanner(System.in);
		int n=kb.nextInt();
		int[] arr=new int[n];
		for(int i=0; i<n; i++){
			arr[i]=kb.nextInt();
		}
		System.out.print(T.solution(n, arr));
	}
}
